frontend/src/components/Forms/EditTailingsReportForm.js

```diff
@@ -59,6 +61,15 @@ export const EditTailingsReportForm = (props) => (
             validate={[]}
           />
         </Form.Item>
+        <Form.Item label="Upload Documents">
+          <Field
+            id="tsf_document_upload"
+            name="tsf_document_upload"
+            expDocumentGuid={props.expDocumentGuid}
+            component={FileUpload}
```

Comment:
Is FileUpload.js specific for tailings? Or can it be reused?

if so, can this be changed to follow the syntax of the other Field Components?

Reply:
Right now it is specific to tailings, as that is the only place where we are currently uploading docs.

The document upload API needs the GUID of the expected document to associate the file to. We could pass in a URL to this component rather than the GUID, but then a higher-level component would have to know and construct the URL. I felt it was better to put it inside this component to make the change footprint smaller.

When you say follow the syntax of other field components, do you mean using another name than expDocumentGuid?

Reply:
I was more so referring to the ```component={FileUpload}```

Can FileUpload be moved into src/components/common and be renamed to match the syntax other similar components? 

Any prop that is passed into to the <Field /> is accessible to FileUpload, it doesn't really matter where this component lives,

I could not have the full picture here, but FileUpload looks as though it can be reused in the future and thus should be kept in common and renamed to `renderFileUpload` for consistency

Reply:
I think moving it to common makes sense as we do intend to use it elsewhere. What should be renamed to renderFileUpload? The component?

Reply:
James re: offline convo with mmumby, please leave the comonent name as FileUpload as it is not specifically a form child component but can be used elsewhere.

Reply:
Took the time to make the component generic. FileUpload now accepts an upload URL rather than a GUID. The form generates the URL from a constant in constants/API.

As well, the tailings page now sends the entire doc down to the children rather than just the GUID, since the expected doc record has all of its related uploaded files which we will eventually be displaying in the modal.

Reply:
Thank you for making it more generic!